packages/yew-router/src/router.rs

```diff
@@ -66,33 +67,30 @@ pub enum Msg<R> {
 pub struct Router<R: Routable + 'static> {
     props: RouterProps<R>,
     #[allow(dead_code)] // only exists to drop listener on component drop
-    route_listener: RouteListener,
-    route: Option<R>,
+    route_listener: EventListener,
 }
 
 impl<R> Component for Router<R>
 where
     R: Routable + 'static,
 {
-    type Message = Msg<R>;
+    type Message = Msg;
     type Properties = RouterProps<R>;
 
     fn create(props: Self::Properties, link: ComponentLink<Self>) -> Self {
-        let route_listener = attach_route_listener(link.callback(Msg::UpdateRoute));
+        let route_listener = EventListener::new(&yew::utils::window(), "popstate", move |_| {
+            link.send_message(Msg::ReRender)
+        });
```

Comment:
Why did you move away from the modular API again?

Reply:
Optimization. The API changed, meaning that in the previous implementation, it'd now read the current route twice. This change ensures the current route is checked and obtained once by the component.

Reply:
That's very unfortunate, are you sure there's no way to avoid that?

Reply:
There will be one when the listener is attached and called and one for the initial render. I'm not aware of any ways that provide a way to handle this.

Reply:
Alright then.